src/pages/Delegate/index.tsx

```diff
@@ -21,14 +21,43 @@ import { Title } from '@/components/ui/title'
 import { DelegateCard } from '@/components/DelegateCard'
 
 export const Delegate = () => {
-  const { api, assetInfo } = useNetwork()
-  const { address } = useParams()
-  const { getConvictionLockTimeDisplay } = useLocks()
+  const { api, assetInfo, selectNetwork } = useNetwork()
+  const { address, incomingNetwork } = useParams()
+  const { selectedAccount } = useAccounts()
+  const getDelegateTx = useGetDelegateTx()
+  const { getConvictionLockTimeDisplay, refreshLocks } = useLocks()
+  const getSubscriptionCallBack = useGetSigningCallback()
+  const navigate = useNavigate()
+  const { search } = useLocation()
+  const [newAddress, setNewAddress] = useState<string>()
+
+  const { getDelegateByAddress, getDelegateByName } = useDelegates()
+
+  useEffect(() => {
+    if (isValidAddress(address!)) {
```

Comment:
Can we do an early return rather than casting, we're not sure that the address will be defined
```suggestion
  if(!address) return

    if (isValidAddress(address)) {
```